### DISCUSSION TITLE
cannot use this model as a layer of other model.

### DISCUSSION THREAD

**Discussion by gormat** [snakers4/silero-models#43] [Q&A] [answered]
this line of code makes it impossible to use this model as a (non trainable) layer of other model. 
https://github.com/snakers4/silero-models/blob/7cf51381e69aeab14da1dd4e324f98fa4ee7b14c/utils.py#L124

also please check discussion here https://discuss.pytorch.org/t/runtimeerror-element-0-of-tensors-does-not-require-grad-and-does-not-have-a-grad-fn-when-training-from-examples/107816/6

I have to manually enable it to make it work. 

Just want to know if there is a specific reason to disable gradient calculation.

**Comment by snakers4** [comment, marked as answer]
> Just want to know if there is a specific reason to disable gradient calculation.

Well, basically these utils were designed mostly with inference in mind
Usually when we write inference code we either disable gradients altogether or use with `with torch.no_grad()`

I do not remember why I went with the former instead of the latter, but probably because I just invoke model simple like `out = model(input)` and I did not want to write an extra function hiding an extra level of tabulation from the concise examples

Anyway you can just override the `init_jit_model` function in your use case and that is it

**Comment by snakers4** [comment]
When github fixes its discussion migration bug I will transfer this issue to a discussion
Please feel free to open another ticket

Will close this for now, I cannot really add anything to this
In any case utils are provided to ease testing and are by no means imperative
